tests/GraphQl/Action/EntrypointActionTest.php

```diff
@@ -81,6 +82,38 @@ public function testPostJsonAction()
         $this->assertEqualsWithoutDateHeader(new JsonResponse(['GraphQL']), $mockedEntrypoint($request));
     }
 
+    public function testFormMultipartAction()
+    {
+        $file = new UploadedFile(
+            __DIR__.'/Fixtures/test.gif',
+            'original.gif',
+            null,
+            UPLOAD_ERR_OK
+        );
+        $request = new Request([], ['operations' => '{"query": "graphqlQuery", "variables": {"file":null}, "operation": "graphqlOperationName"}', 'map' => '{"0":["variables.file"]}'], [], [], ['0' => $file]);
+        $request->setMethod('POST');
+        $request->headers->set('Content-Type', 'multipart/form-data');
+
+        $schema = $this->prophesize(Schema::class);
+        $schemaBuilderProphecy = $this->prophesize(SchemaBuilderInterface::class);
+        $schemaBuilderProphecy->getSchema()->willReturn($schema->reveal());
+
+        $executionResultProphecy = $this->prophesize(ExecutionResult::class);
+        $executionResultProphecy->toArray(3)->willReturn(['GraphQL']);
+        $executorProphecy = $this->prophesize(ExecutorInterface::class);
+        $executorProphecy->executeQuery(Argument::is($schema->reveal()), 'graphqlQuery', null, null, ['file' => $file], 'graphqlOperationName')->willReturn($executionResultProphecy->reveal());
+
+        $twigProphecy = $this->prophesize(TwigEnvironment::class);
+        $routerProphecy = $this->prophesize(RouterInterface::class);
+
+        $graphiQlAction = new GraphiQlAction($twigProphecy->reveal(), $routerProphecy->reveal(), true);
+        $graphQlPlaygroundAction = new GraphQlPlaygroundAction($twigProphecy->reveal(), $routerProphecy->reveal(), true);
+
+        $mockedEntrypoint = new EntrypointAction($schemaBuilderProphecy->reveal(), $executorProphecy->reveal(), $graphiQlAction, $graphQlPlaygroundAction, true, true, true, 'graphiql');
+
+        $this->assertEqualsWithoutDateHeader(new JsonResponse(['GraphQL']), $mockedEntrypoint($request));
```

Comment:
Shouldn't you do make more tests (particularly on the map to variables)?

Reply:
I'll do that.

Reply:
I added couple of tests but about the map to variables, do you think we should return error if map does not match the variables?
for example:
map:
`json
{
      "file":["variables.file"]
}
`
variables:
`json
"variables":{
       "something": null
}
`

Reply:
Yes it could be a good idea :slightly_smiling_face: